ml-dsa/src/lib.rs

```diff
@@ -0,0 +1,10 @@
+use std::cell::RefCell;
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+use crate::constants::{F, D};
+
+thread_local! {
+    pub static CURRENT_BUILDER: RefCell<Option<*mut CircuitBuilder<F, D>>> = RefCell::new(None);
+}
```

Comment:
Nice pattern! I'm a bit hesitant to introduce unsafe code here, but I think it's fine* in this case